src/Device/Driver/LX/NanoLogger.cpp

```diff
@@ -219,25 +219,22 @@ Nano::ReadFlightList(Port &port, RecordedFlightList &flight_list,
     return nflights == 0;
 
   env.SetProgressRange(nflights);
-
-  unsigned requested_tail = 1;
+  
+  unsigned requested_tail = 0;
   while (true) {
-    const unsigned room = flight_list.max_size() - flight_list.size();
-    const unsigned remaining = nflights - requested_tail + 1;
-    const unsigned nmax = std::min(room, remaining);
-    if (nmax == 0)
-      break;
 
-    /* read 8 records at a time */
-    const unsigned nrequest = std::min(nmax, 8u);
+    if (nflights == 0)
+      break;
 
     timeout = TimeoutClock(2000);
     if (!GetLogbookContents(port, reader, flight_list,
-                            requested_tail, nrequest, env, timeout))
+                            nflights, 1, env, timeout))
```

Comment:
Previously, XCSoar did bulk transfer, and now you're doing one at a time (but you're still using the function for bulk transfers, just passing `n=1`). This may reduce performance and affects how timeouts are handled. Why?

Reply:
It was easier for me with my basic coding skills to implement the change this way.
I have not experienced any timeouts or noticeable performance reductions to date.
If required I'll have a go at reinstating the change using bulk transfers of 8 at a time.